### DISCUSSION TITLE
Discord-like roles

### DISCUSSION THREAD

**Discussion by chetanyakan** [mattermost/community#95] [Enhancements]
A plugin to allow custom roles in Mattermost.
Can show the roles in user profile popover.
Using roles, custom permissions can be granted to specific users like auto-join/leave specific private channels.
Ability to post in custom channels.

**Comment by esethna** [comment]
We currently show the "team admin" and "sys admin" badges in the pop-over but having more customized roles is certainly something we've been thinking about for a while

**Reply by chetanyakan** [reply]
So, an interesting idea came up in another discussion: https://github.com/mattermost/community/discussions/140#discussioncomment-920700

We can use the membership for a (private) channel to allow plugins to grant some custom roles.

**Reply by chetanyakan** [reply]
We can set the channel description to something like:

> All those having access to this channel will be able to use the demo plugin hidden features. For more info, type `/demo help`
